wagtail/contrib/search_promotions/views.py

```diff
@@ -125,121 +125,113 @@ def save_searchpicks(query, new_query, searchpicks_formset):
         return False
 
 
-@permission_required("wagtailsearchpromotions.add_searchpromotion")
-def add(request):
-    if request.method == "POST":
-        # Get query
-        query_form = forms.QueryForm(request.POST)
-        if query_form.is_valid():
-            query = Query.get(query_form["query_string"].value())
-
-            # Save search picks
-            searchpicks_formset = forms.SearchPromotionsFormSet(
-                request.POST, instance=query
+class AddSearchPromotion(generic.CreateView):
```

Comment:
This is missing the `permission_policy`, and it would also be nice if we can show the breadcrumbs here